presto-spi/src/main/java/com/facebook/presto/spi/type/Type.java

```diff
@@ -83,26 +84,56 @@
      */
     boolean getBoolean(Block block, int position);
 
+    /**
+     * Gets the value at the {@code block} {@code position - offsetBase} as a boolean
+     * without boundary checks.
+     */
+    boolean getBooleanUnchecked(UncheckedBlock block, int internalPosition);
```

Comment:
It's not necessary because whereas `getOffsetBase` behaves consistently across all Blocks, whereas `Type#getXXXUnchecked` are meant to delegate to some internal representation.  For example, `#getBooleanUnchecked` delegates to `Block#getByteUnchecked`, `#getLongUnchecked` internally delegates to a wide host of methods--long, int, short, byte, depending on the type.